pkgs/development/libraries/gettext/default.nix

```diff
@@ -11,6 +11,12 @@ stdenv.mkDerivation rec {
   patches = [
     ./absolute-paths.diff
     ./gettext.git-2336451ed68d91ff4b5ae1acbc1eca30e47a86a9.patch
+  ]
+  # HACK: Since this is merely a UX fix, to minimise rebuilds only patch
+  #       the final library.
+  #       Remove the entire thing when updating to the next release.
+  ++ lib.optionals (stdenv.isDarwin && !(lib.hasPrefix "bootstrap" stdenv.name)) [
+    ./gettext.git-ec0e6b307456ceab352669ae6bccca9702108753.patch
```

Comment:
can you use fetchpatch to avoid adding unnecessary files to nixpkgs?

Reply:
Sure, I just thought I’d do the same as the other existing patch.

Reply:
Fair enough. The best practice is to keep the patches out of tree when possible but not it's clearly not uniformly enforced :)